github/status.go

```diff
@@ -135,61 +134,10 @@ func (s status) Add(sha, state, targetURL, description string) error {
 		// Happy path
 		return nil
 	case http.StatusNotFound:
-		msg := disambiguateError(s)
+		msg := "The user with this token doesn't have write access to the repo or the repo doesn't exists (less likely)"
```

Comment:
Doing an http request `API: GET /repos/:owner/:repo` and expecting to get status code 200 requires different token scope than `repo:status`. Hence there is no point to keep all of this code logic if we will never change the token scope.
Additionally, even if we change the scope to `repo`, if a does not have at least a read access to the repo we still get 404 status code (i.e with `pix4d-service-account-cloud` account) and we can still not distinguish which case we have.

Reply:
I am lost here. I forgot the details, but the code was able to disambiguate. Are you saying that in the tests I was using a token with broader scope ?

Reply:
> Are you saying that in the tests I was using a token with broader scope ?

yes.

> I am lost here. I forgot the details, but the code was able to disambiguate. 

It is if you change the token scope.

Reply:
Note that in my test repo https://github.com/Pix4D/test-cogito I am the admin, but my token has only `repo:status` scope and I get 404 here.

Reply:
Quick check:
`curl -X GET -H "Accept: application/vnd.github.v3+json" -H "Authorization: token here_goes_the_token" https://api.github.com/repos/pix4d/your_repo_name`

and try changing the scope.

Reply:
As three things could be going wrong here, I would word the message differently:
```suggestion
		msg := "Either the user and/or its token don't have the right permission on the repo or the repo doesn't exists."+
			" (User must have write access on this repo and token must have scope repo:status)"
```

Reply:
> Note that the check performed here was very biased towards Pix4D-Janus which has a read access by default and it probably had a token before with full scope.

Ah, I think that's it. It could explain this, good catch.

Reply:
I propose the following as message, which conveys all possible cases in a format that is clear for the end user:

```
One of the following happened:

1. The repo <REPO> doesn't exist.
2. The user who issued the token doesn't have write access to the repo.
3. The token doesn't have scope repo:status.
```